pkg/apiserver/experiment/experiment.go

```diff
@@ -16,17 +16,40 @@ package experiment
 import (
 	"context"
 	"net/http"
+	"sync"
+	"time"
 
 	"github.com/gin-gonic/gin"
-
 	"github.com/pingcap/chaos-mesh/api/v1alpha1"
 	statuscode "github.com/pingcap/chaos-mesh/pkg/apiserver/status_code"
 	"github.com/pingcap/chaos-mesh/pkg/config"
 	"github.com/pingcap/chaos-mesh/pkg/core"
+	"golang.org/x/sync/errgroup"
 
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+var (
+	// TODO(yeya24): don't hardcode this if it is possible
+	kinds = map[string]v1alpha1.ChaosList{
```

Comment:
we can register these with `init` function in `v1alpha1` pkg.

Reply:
In the v1aloha1 package? Then this map will be public, which is not very appropriate I think. It is error-prone and maybe changed in the runtime. What’s your idea?

Reply:
It can reduce the cost of adding new chaos. I think it is also OK to do so at present like your pr.

Reply:
You are right, we should put it in a commonplace and use init(). But I am not sure in `v1alpha1` package or in `apiserver`.